src/Hosting/Server.Abstractions/src/IHostContextContainer.cs

```diff
@@ -13,6 +13,6 @@ public interface IHostContextContainer<TContext> where TContext : notnull
         /// <summary>
         /// Represents the <typeparamref name="TContext"/>  of the host.
         /// </summary>
-        TContext HostContext { get; set; }
+        TContext? HostContext { get; set; }
```

Comment:
Its a little weird that TContext has generic constraint of notnull, but the only place it is used is nullable.

Is notnull a way of saying "this type can be anything but `Nullable<Struct>`"

Reply:
I think this is the way it is because if flows from `IHttpApplication<TContext>` which has the same constraint. You can use your own implementation of that if you call IServer.StartAsync directly, and we probably don't want to have to deal with `IHttpApplication<HttpContext?>` and the like.